backuptar/tar.go

```diff
@@ -109,6 +109,69 @@ func BasicInfoHeader(name string, size int64, fileInfo *winio.FileBasicInfo) *ta
 	return hdr
 }
 
+// EncodeSDDLFromTarHeader reads the SDDL associated with the header of the current file
+// from the tar header and encodes it into a byte slice.
+func EncodeSDDLFromTarHeader(hdr *tar.Header) ([]byte, error) {
+	// Maintaining old SDDL-based behavior for backward
+	// compatibility.  All new tar headers written by this library
+	// will have raw binary for the security descriptor.
+	var sd []byte
+	var err error
+	if sddl, ok := hdr.PAXRecords[hdrSecurityDescriptor]; ok {
+		sd, err = winio.SddlToSecurityDescriptor(sddl)
+		if err != nil {
+			return nil, err
+		}
+	}
+	if sdraw, ok := hdr.PAXRecords[hdrRawSecurityDescriptor]; ok {
+		sd, err = base64.StdEncoding.DecodeString(sdraw)
+		if err != nil {
+			return nil, err
+		}
+	}
+	return sd, nil
+}
+
+// EncodeExtendedAttributesFromTarHeader reads the EAs associated with the header of the
+// current file from the tar header and encodes it into a byte slice.
+func EncodeExtendedAttributesFromTarHeader(hdr *tar.Header) ([]byte, error) {
```

Comment:
Prefixing these functions with `Encode` seems a bit odd to me, especially since we are really parsing or decoding data from the tar header. What do you think about just dropping `Encode` from the names (e.g. `ExtendedAttributesFromTarHeader`)?

Reply:
I had the `Encode` prefix because we return the encoded attributes or security descriptors but I agree that the primary job of this function is not encoding things. I have dropped the prefix.